### PR TITLE
Add unit tests for com.alibaba.nacos.client.utils.StringUtils

### PR DESCRIPTION
**Pull request by EricHetti**
I've analysed your codebase and noticed that com.alibaba.nacos.client.utils.StringUtils is not fully tested.
I've written some tests for the methods in this class with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests will help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other classes that you consider important.

### CODE REVIEW

File: client/src/test/java/com/alibaba/nacos/client/StringUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    public void testEquals() {
+        assertTrue(StringUtils.equals("foo", "foo"));
+        assertFalse(StringUtils.equals("bar", "foo"));
+    }
+
+    @Test
+    public void testSubstringBetween() {
+        assertNull(substringBetween(null, null, null));
+
+        assertEquals(null, substringBetween("", "foo", ""));
```
Comment (JohnLBergqvist) [inline client/src/test/java/com/alibaba/nacos/client/StringUtilsTest.java]: You could replace this with `assertNull` again.

**Review by JohnLBergqvist — CHANGES_REQUESTED** [review]

File: client/src/test/java/com/alibaba/nacos/client/StringUtilsTest.java (hunk 1 above)
Comment (Braavos6) [reply client/src/test/java/com/alibaba/nacos/client/StringUtilsTest.java]: Missed that. Thanks I changed it

**Review by JohnLBergqvist — APPROVED** [review]

**Review by paulkennethkent — APPROVED** [review]

File: client/src/test/java/com/alibaba/nacos/client/StringUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        assertTrue(StringUtils.equals("foo", "foo"));
+        assertFalse(StringUtils.equals("bar", "foo"));
+    }
+
+    @Test
+    public void testSubstringBetween() {
+        assertNull(substringBetween(null, null, null));
+        assertNull(substringBetween("", "foo", ""));
+
+        assertEquals("", substringBetween("foo", "foo", ""));
```
Comment (peterschrammel) [inline client/src/test/java/com/alibaba/nacos/client/StringUtilsTest.java:50]: Maybe one more without empty string?

File: client/src/test/java/com/alibaba/nacos/client/StringUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    @Test
+    public void testDefaultIfEmpty() {
+        assertEquals("foo", defaultIfEmpty("", "foo"));
+        assertEquals("bar", defaultIfEmpty("bar", "foo"));
+    }
+
+    @Test
+    public void testEquals() {
+        assertTrue(StringUtils.equals("foo", "foo"));
```
Comment (peterschrammel) [inline client/src/test/java/com/alibaba/nacos/client/StringUtilsTest.java:37]: What happens with null?

**Review by peterschrammel — APPROVED** [review]

File: client/src/test/java/com/alibaba/nacos/client/StringUtilsTest.java (hunk 2 above)
Comment (Braavos6) [reply client/src/test/java/com/alibaba/nacos/client/StringUtilsTest.java:50]: added test case

File: client/src/test/java/com/alibaba/nacos/client/StringUtilsTest.java (hunk 3 above)
Comment (Braavos6) [reply client/src/test/java/com/alibaba/nacos/client/StringUtilsTest.java:37]: added test case

File: client/src/test/java/com/alibaba/nacos/client/StringUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import static com.alibaba.nacos.client.utils.StringUtils.*;
+
+public class StringUtilsTest {
+
+    @Test
+    public void testisNotBlank() {
+        assertTrue(isNotBlank("foo"));
+
+        assertFalse(isNotBlank(" "));
+        assertFalse(isNotBlank(null));
```
Comment (peterschrammel) [inline client/src/test/java/com/alibaba/nacos/client/StringUtilsTest.java:19]: What about `"", "\t", "\n"`?

**Review by peterschrammel — APPROVED** [review]

File: client/src/test/java/com/alibaba/nacos/client/StringUtilsTest.java (hunk 4 above)
Comment (Braavos6) [reply client/src/test/java/com/alibaba/nacos/client/StringUtilsTest.java:19]: I didn't put them in as there was 100% coverage, but I can put them in if you like :)

**Review by peterschrammel — APPROVED** [review]

**Pull request closed without merging**